crates/common/lean_chain/Cargo.toml

```diff
@@ -0,0 +1,24 @@
+[package]
+name = "ream-lean-chain"
+authors.workspace = true
+edition.workspace = true
+keywords.workspace = true
+license.workspace = true
+readme.workspace = true
+repository.workspace = true
+rust-version.workspace = true
+version.workspace = true
+
+[dependencies]
+alloy-primitives.workspace = true
+serde.workspace = true
+ssz_types.workspace = true
+tokio.workspace = true
+tracing.workspace = true
+tree_hash.workspace = true
+
+# ream dependencies
+ream-consensus-lean.workspace = true
+ream-network-spec.workspace = true
+ream-p2p.workspace = true
+ream-pqc.workspace = true
```

Comment:
should we structure this like our other crates like common/consensus/*, maybe common/chain/*, so this crate would become ream-chain-lean, and the beacon one would be ream-chain-beacon